layouts/_default/rss.xml

```diff
@@ -0,0 +1,64 @@
+{{- $pctx := . -}}
+{{- if .IsHome -}}{{ $pctx = .Site }}{{- end -}}
+{{- $pages := slice -}}
+{{- if or $.IsHome $.IsSection -}}
+{{- $pages = $pctx.RegularPages -}}
+{{- else -}}
+{{- $pages = $pctx.Pages -}}
+{{- end -}}
+{{- $limit := .Site.Config.Services.RSS.Limit -}}
+{{- if ge $limit 1 -}}
+{{- $pages = $pages | first $limit -}}
+{{- end -}}
+{{- printf "<?xml version=\"1.0\" encoding=\"utf-8\" standalone=\"yes\"?>" | safeHTML }}
+<rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom">
+  <channel>
+    <title>{{ if eq  .Title  .Site.Title }}{{ .Site.Title }}{{ else }}{{ with .Title }}{{.}} on {{ end }}{{ .Site.Title }}{{ end }}</title>
+    
+    {{ if .Site.Params.RssImage }}<image>
+      <url>{{ printf "%s%s" .Permalink .Site.Params.RssImage }}</url>
+      <title>{{ if eq  .Title  .Site.Title }}{{ .Site.Title }}{{ else }}{{ with .Title }}{{.}} on {{ end }}{{ .Site.Title }}{{ end }}</title>
+      <link>{{ .Permalink }}</link>
+    </image>
+    {{ end }}
+    
+    <link>{{ .Permalink }}</link>
+    <description>Recent content {{ if ne  .Title  .Site.Title }}{{ with .Title }}in {{.}} {{ end }}{{ end }}on {{ .Site.Title }}</description>
+    <generator>Hugo -- gohugo.io</generator>{{ with .Site.LanguageCode }}
+    <language>{{.}}</language>{{end}}{{ with .Site.Author.email }}
+    <managingEditor>{{.}}{{ with $.Site.Author.name }} ({{.}}){{end}}</managingEditor>{{end}}{{ with .Site.Author.email }}
+    <webMaster>{{.}}{{ with $.Site.Author.name }} ({{.}}){{end}}</webMaster>{{end}}{{ with .Site.Copyright }}
+    <copyright>{{.}}</copyright>{{end}}{{ if not .Date.IsZero }}
+    <lastBuildDate>{{ .Date.Format "Mon, 02 Jan 2006 15:04:05 -0700" | safeHTML }}</lastBuildDate>{{ end }}
+    {{ with .OutputFormats.Get "RSS" }}
+	{{ printf "<atom:link href=%q rel=\"self\" type=%q />" .Permalink .MediaType | safeHTML }}
+    {{ end }}
+    {{ range $pages }}
+    <item>
+      <title>{{ .Title }}</title>
+      <link>{{ .Permalink }}</link>
+      <pubDate>{{ .Date.Format "Mon, 02 Jan 2006 15:04:05 -0700" | safeHTML }}</pubDate>
+      {{ with .Site.Author.email }}<author>{{.}}{{ with $.Site.Author.name }} ({{.}}){{end}}</author>{{end}}
+      <guid>{{ .Permalink }}</guid>
+      <description>
+        {{ if .Params.Cover }}
```

Comment:
I use Feedly, too! :) I think there's a lot to unwrap here about images in feeds and Feedly. I've had to rewrite this twice just to try and keep things simple, so hopefully this all makes sense.

### XML <image> tag for RSS
I had to read up on W3 for this stuff as well, and that `<image>` tag is for the RSS channel as a whole. It does not display images inside the content (which is all HTML format held inside the `<description>` tag). I did add an XML `<image>` tag near the top of the RSS generator, but I'm unsure if it was successful. (see Feedly update issues)

### Feedly cover images in list view (Part 1 - why they're still missing)
Feedly will not update existing feed content, even if the RSS data details change. All old post data will be displayed the same forever. If it had no cover image before, it won't update to include one. At least, with all my testing I haven't found a single way to update my feed, and it's really annoying. Only newly published items seem to have their data read fresh. I think I will create a test site with an RSS feed that we can publish stuff to and see our various code updates.

### Feedly cover images in list view (Part 2 - they aren't RSS-based)
The post image in Feedly's list view doesn't actually come from the RSS data. (At least, I'm 95% sure of this.) It comes from the Hugo-generated [Open Graph data here](https://github.com/gohugoio/hugo/blob/25a6b33693992e8c6d9c35bc1e781ce3e2bca4be/tpl/tplimpl/embedded/templates/opengraph.html#L5-L16). So if that data isn't present in your generated pages because your image data hasn't been found by Hugo, the image won't show up in the list view

![Screen Shot 2020-05-26 at 14 28 02](https://user-images.githubusercontent.com/1312251/82942460-9a6f9400-9f5d-11ea-9040-7bcca295ce0e.png)
(Coincidentally, this is also the incorrect cover image, because it wouldn't update for me.)

The cover image data that I added into the RSS feed (which is just an HTML `<img>` inside the XML `<description>` tag) appears at the top of the actual content view like this.

![Screen Shot 2020-05-26 at 14 33 49](https://user-images.githubusercontent.com/1312251/82943948-0226de80-9f60-11ea-88f0-13ca1ca71888.png)

I'm a little worried that other feed readers will not display things the same way (or possibly at all), but I'm at least sure this is what should occur in Feedly.

Check out my RSS feed for adamormsby.com. If you haven't looked at my feed in Feedly yet, you may actually get the most updated version of my site's output. Perhaps it will show you what I mean. (Also, I'd love to see it if you do!)

**TL;DR:** I think all my stuff is correct, but it's worth testing again on a new test site.

Do these explanations help?

Reply:
Also, I added you to a small testbed. Feel free to run any tests and push any new posts to see if everything works as expected. Hopefully, we can sort out what's working, what isn't, and what's Feedly's fault. 😆

Reply:
Whoa, great explanation 👍 I've just tested it again with my other Hugo based sites and you're 100% right. `Image` is not RSS based and it's a little fucked up in Feedly 😅

My only concern is if we should keep relative paths here https://github.com/panr/hugo-theme-hello-friend/pull/148/files#diff-cd5b9b2f01fe87b55aab7340a230e838R45 and here  https://github.com/panr/hugo-theme-hello-friend/pull/148/files#diff-cd5b9b2f01fe87b55aab7340a230e838R51 🤔 I spotted this issue in some random free RSS viewers like https://feed.mikle.com, where the paths incorrectly direct to the `RSS viewer domain` + `/relative path to the image`

Reply:
Hmm... I'm not sure I understand. How is the RSS domain getting into the URL at all? Hugo builds everything locally into the `public/index.xml` file, and I'm seeing correct full domain links in mine. Here's some of my output for that section:

`&lt;img src=&#34;https://www.adamormsby.com/posts/hugo-themes-making-modifications/cover.jpg&#34;/&gt;`
(The escaped characters convert later, of course.)

Could it be that the RSS viewer is modifying them when it pulls the site's RSS data?

Reply:
Hmm... maybe it's something with Vercel, but check deployed test site and https://hello-friend.now.sh/index.xml. The URLs for the images (I'm using `<img src="{{ .Params.Cover | absURL }}" class="post-cover" />`) are somehow relative 🤔 I think something is messed up with `baseURL` on the Vercel side ¯\_(ツ)_/¯

I see that your site has proper URLs... So we can leave it as it is for now 😉

Reply:
Hmm yeah, could be. I've certainly shot myself in the foot before by forgetting to update my baseURL. Those kinds of issues can hide for days!